Menubar bug fix

hoelzro opened the pull request.

hoelzro committed:
Convert menubar item to use conditional shortcut syntax

hoelzro committed:
Honor custom-menu-content when is-dropdown is set in menubar plugin

Fixes GH #9040

pmario commented on plugins/tiddlywiki/menubar/menu.tid:
```diff
@@ -7,34 +7,37 @@ tags: $:/tags/PageTemplate
+	<$transclude field="custom-menu-content" mode="inline"/>
+	<%else%>
 	<$set name="tv-wikilinks" value="no">
 	<$transclude field="caption" mode="inline"/>
```
If there is no caption field, it should show the title.

hoelzro replied:
Is that what the old code does?  When I create a tiddler with no `caption`, I see a menubar item with no text in it.

pmario commented on plugins/tiddlywiki/menubar/menu.tid:
```diff
@@ -7,34 +7,37 @@ tags: $:/tags/PageTemplate
 <$list filter="[[$size$]match[wide]] ~[<currentTiddler>get[show-when]match[$size$]] ~[{$:/state/popup/menubar/hamburger}match[yes]]" variable="ignore">
 <li style={{!!custom-menu-styles-$size$}} class={{{ [<currentTiddler>get[show-when]addprefix[tc-menubar-]] tc-menubar-item +[join[ ]] }}}>
-<$list filter="[<currentTiddler>!is-dropdown[yes]]" variable="listItem" emptyMessage="""
+<%if [<currentTiddler>!is-dropdown[yes]] %>
```
I would prefer "positive" logic. So IMO we should use `is-dropdown[yes]` instead of `!is-dropdown`

pmario replied:
`[<currentTiddler>is-dropdown[yes]]` should be good, so imo compare is not needed

pmario commented on plugins/tiddlywiki/menubar/menu.tid:
```diff
@@ -7,34 +7,37 @@ tags: $:/tags/PageTemplate
+<%if [<currentTiddler>has[custom-menu-content]] %>
+	<$transclude field="custom-menu-content" mode="inline"/>
+<%else%>
+	<!-- Link -->
```
IMO this comment id redundant, since the next widget is `$link`

hoelzro replied:
I was mostly preserving the existing code, but I agree that it's redundant.

hoelzro commented:
@pmario Thanks for the feedback! I removed that link comment and changed the conditional; we just need to decide what to do if `caption` is missing now.

hoelzro committed:
Fall back to showing title in menubar in case of no caption

hoelzro commented:
@pmario It took me a while, but I just added a fix for that too!
